docs/tutorial/go.rst

```diff
@@ -0,0 +1,361 @@
+install ``go``.
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:install-go]
+    :end-before: [docs:install-go-end]
+    :dedent: 2
+
+Initialise the new Go module:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:mod-init-go]
+    :end-before: [docs:mod-init-go-end]
+    :dedent: 2
+
+Create a ``main.go`` file, copy the following text into it and then
+save it:
+
+.. literalinclude:: code/go/main.go
+    :language: go
+
+Build the Go application so it can be run:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:go-build]
+    :end-before: [docs:go-build-end]
+    :dedent: 2
+
+A new binary named ``go-hello-world`` should be in the current
+directory. This binary is only needed for local testing, as
+Rockcraft will compile the Go application. Run the Go application using
+``./go-hello-world`` to verify that it works.
+
+The code in ``main.go`` starts up an HTTP server listening on port 8000
+that we can test by using ``curl`` to send a request to the root
+endpoint. We may need a new terminal for this, if we are using Multipass use
+``multipass shell rock-dev`` to get another terminal:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:curl-go]
+    :end-before: [docs:curl-go-end]
+    :dedent: 2
+
+The Go application should respond with ``Hello, world!``.
+
+The Go application looks good, so stop it for now using :kbd:`Ctrl` + :kbd:`C`.
+
+Pack the Go application into a rock
+===================================
+
+
+First, we'll need a ``rockcraft.yaml`` file. Rockcraft will automate its
+creation and tailor it for a Go application by using the
+``go-framework`` profile:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:create-rockcraft-yaml]
+    :end-before: [docs:create-rockcraft-yaml-end]
+    :dedent: 2
+
+The ``rockcraft.yaml`` file will automatically be created in the project working
+directory. Open it in a text editor and check that the ``name`` is
+``go-hello-world``. Ensure that ``platforms`` includes the architecture of
+the host. For example, if the host uses the ARM
+architecture, include ``arm64`` in ``platforms``.
+
+.. note::
+    For this tutorial, we'll use the ``name`` ``go-hello-world`` and assume
+    we are running on an ``amd64`` platform. Check the architecture of the
+    system using ``dpkg --print-architecture``.
+
+    The ``name``, ``version`` and ``platform`` all influence the name of the
+    generated ``.rock`` file.
+
+Pack the rock:
+
+.. literalinclude:: code/go/task.yaml
+    :language: bash
+    :start-after: [docs:pack]
+    :end-before: [docs:pack-end]
+    :dedent: 2
+
+.. note::
+
+    Depending on the network, this step can take a couple of minutes to finish.
+
+    ``ROCKCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS`` is required whilst the ``go-framework``
+    extension is experimental.
```

Comment:
If this is still accurate, then we urgently need to make this a separate step. Dirtying the `rockcraft pack` step with a shell var declaration diminishes its importance.

Reply:
I have updated the tutorial so the exported envvar `ROCKCRAFT_ENABLE_EXPERIMENTAL_EXTENSIONS` is added in the first part of the tutorial, so it is not added to the `rockcraft pack` commands.

Reply:
Looks good, thanks!